### PR TITLE
PR #3 Fixes

### PR DESCRIPTION
**Pull request by scripttease**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -16,17 +16,11 @@ scalacOptions ++= Seq(
 )
 
 /* deps for aws lambda */
-libraryDependencies ++= Seq(
-  "com.amazonaws" % "aws-lambda-java-core" % "1.1.0",
-  "com.gu" %% "content-api-client" % "11.22",
-  "org.typelevel" %% "cats" % "0.9.0",
+libraryDependencies += "com.amazonaws" % "aws-lambda-java-core" % "1.1.0"
```
Comment (scripttease) [inline build.sbt]: The boilerplate for the lambda object (which I am not currently using but hope to be soon) will not compile without this dep so I have put it back in

File: src/main/scala/com/gu/kindlegen/Lambda.scala [hunk 2]
```diff
@@ -62,174 +48,3 @@ object TestIt {
...
-    val response = Await.result(capiClient.getResponse(query), 5.seconds)
-    response.results
-  }
-
-  def resultToArticles(response: Seq[com.gu.contentapi.client.model.v1.Content]): Seq[Article] = {
-    response.map(responseContent =>
-      Article(responseContent))
-  }
-
-  def toBookSectionPageList(articles: List[Article]): List[BookSectionPage] = {
```
Comment (scripttease) [inline src/main/scala/com/gu/kindlegen/Lambda.scala]: This method has been refactored and moved to the BookSectionPage companion object in `BookSectionPage.scala`

File: src/main/scala/com/gu/kindlegen/Lambda.scala [hunk 3]
```diff
@@ -62,174 +48,3 @@ object TestIt {
...
-    // TODO: Query requires use-date and exact date of publication
-    val response = Await.result(capiClient.getResponse(query), 5.seconds)
-    response.results
-  }
-
-  def resultToArticles(response: Seq[com.gu.contentapi.client.model.v1.Content]): Seq[Article] = {
-    response.map(responseContent =>
-      Article(responseContent))
-  }
-
```
Comment (scripttease) [inline src/main/scala/com/gu/kindlegen/Lambda.scala]: Querier object has been moved to `Querier.scala` along with the `resultToArticles` method, now renamed as `responseToArticles` for clarity

File: src/main/scala/com/gu/kindlegen/Lambda.scala [hunk 4]
```diff
@@ -62,174 +48,3 @@ object TestIt {
...
-    val lastBookSection = BookSection(
-      bookSectionId = lastBookSectionPages.head.bookSectionId,
-      bookSectionTitle = lastBookSectionPages.head.articles.head.sectionName,
-      pages = lastBookSectionPages.reverse
-    )
-    val r = (lastBookSection :: chunkedPages._1).reverse
-      .map(x => Tuple2(x.pages.length, List(x.pages.map(_.bookSectionId))))
-    println(r)
-    (lastBookSection :: chunkedPages._1).reverse
-  }
```
Comment (scripttease) [inline src/main/scala/com/gu/kindlegen/Lambda.scala]: This method has been refactored and moved to the BookSection companion object in BookSection.scala

File: src/main/scala/com/gu/kindlegen/Lambda.scala [hunk 5]
```diff
@@ -62,174 +48,3 @@ object TestIt {
...
-
-  def toSectionHeading(articles: Seq[Article]): Seq[SectionHeading] = {
-    articles.map(x =>
-      SectionHeading(
-        title = x.sectionName,
-        titleLink = x.newspaperBookSection + ".xml"
-      ))
-  }
```
Comment (scripttease) [inline src/main/scala/com/gu/kindlegen/Lambda.scala]: This method has been moved to the SectionManifest companion object in `SectionManifest.scala`

File: src/main/scala/com/gu/kindlegen/Lambda.scala [hunk 6]
```diff
@@ -62,174 +48,3 @@ object TestIt {
...
-
-  // Pass in Querier.resultToArticles(getPrintSentResponse)
-  def toManifest(articles: Seq[Article], buildDate: DateTime = DateTime.now): SectionManifest = {
-    SectionManifest(
-      publicationDate = articles.head.pubDate,
-      buildDate = buildDate,
-      sections = Querier.toSectionHeading(articles)
-    )
-  }
-}
```
Comment (scripttease) [inline src/main/scala/com/gu/kindlegen/Lambda.scala]: This method has been moved and refactored to the apply method SectionManifest companion object in `SectionManifest.scala`

File: src/test/scala/com/gu/kindlegen/LambdaSuite.scala [hunk 7]
```diff
@@ -20,769 +12,4 @@ class LambdaSuite extends FunSuite {
...
-    val r = expectedOutBS2.map(x => Tuple2(x.pages.length, List(x.pages.map(_.bookSectionId))))
-    println(s"expected: $r")
-    assert(Querier.toBookSectionList(expectedOutBSP2) === expectedOutBS2)
-  }
-
-  test(".toBookSectionList with one page") {
-    assert(Querier.toBookSectionList(expectedOutBSP1) === expectedOutBS1)
-  }
 }
-
```
Comment (scripttease) [inline src/test/scala/com/gu/kindlegen/LambdaSuite.scala]: Writing the TestContent class rendered most of these test articles created above redundant/much simpler. The majority of the functionality tested above has also been moved to other classes for clarity, as described in the comments above

File: src/main/scala/com/gu/kindlegen/Article.scala [hunk 8]
```diff
@@ -23,7 +23,7 @@ case class Article(
 )
 
 object Article {
-  def apply(content: Content) = new Article(
+  def apply(content: Content): Article = new Article(
```
Comment (annebyrne) [inline src/main/scala/com/gu/kindlegen/Article.scala]: 👍

File: src/main/scala/com/gu/kindlegen/BookSection.scala [hunk 9]
```diff
@@ -9,3 +9,20 @@ case class BookSection(bookSectionId: String, bookSectionTitle: String, pages: L
   def numPages = pages.size
   def numArticles = pages.flatMap(_.articles).size
 }
+
+object BookSection {
+
+  def chunkBookSectionPages(pages: List[BookSectionPage]): List[List[BookSectionPage]] = {
+    val tsil = pages.foldLeft(List.empty[List[BookSectionPage]]) {
```
Comment (annebyrne) [inline src/main/scala/com/gu/kindlegen/BookSection.scala]: `List.empty` doesn't seem the most idiosyncratic to me. Why is it required here? A more descriptive variable name here would also be helpful.

File: src/main/scala/com/gu/kindlegen/BookSection.scala [hunk 10]
```diff
@@ -9,3 +9,20 @@ case class BookSection(bookSectionId: String, bookSectionTitle: String, pages: L
...
+      // case where the id of the first page in the first list is the same as current enum page id.
+      case ((head :: tail), page) if head.map(_.bookSectionId).head == page.bookSectionId => {
```
Comment (annebyrne) [inline src/main/scala/com/gu/kindlegen/BookSection.scala]: The two uses of `.head` here make this code a little hard to follow. Is there a better way to extrapolate out the useful values to compare against in the list?

File: src/main/scala/com/gu/kindlegen/BookSectionPage.scala [hunk 11]
```diff
@@ -7,4 +7,26 @@ case class BookSectionPage(bookSectionId: String, pageNum: Int, articles: List[A
 
 // TODO: Ask David B if I should group by BookSectionPage as well as page number - as in, can a page have two book sections in?
 // TODO: allow edge case where a page can have more than one book section on it.
+// TODO: Ask DB if the order INSIDE the pages matters (ie order of the articles) and if so how I recreate this.
 
+//
+object BookSectionPage {
+  def chunkByPageNum(articles: List[Article]): List[List[Article]] = {
+    val tsil = articles.foldLeft(List.empty[List[Article]]) {
```
Comment (annebyrne) [inline src/main/scala/com/gu/kindlegen/BookSectionPage.scala]: Similar comments to above.

File: src/test/scala/com/gu/kindlegen/QuerierSpec.scala [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+  // TODO: Find a way to test printSentResponse, extract the edition dates etc
+  def formatter = DateTimeFormat.forPattern("yyyyMMdd")
+  val capiDate = CapiModelEnrichment.RichJodaDateTime(formatter.parseDateTime("20170724")).toCapiDateTime
+  val testcontent = TestContent("", "", 3, "", "", capiDate, capiDate, capiDate, "", "", "").toContent
+  val capiResponse = List(testcontent)
+
+  ".responseToArticles" should "convert a capi response (Seq[Content) to a Seq[Article]" in {
+    val toArticles = Querier.responseToArticles(capiResponse)
+    assert(toArticles.head.newspaperPageNumber === 3)
```
Comment (annebyrne) [inline src/test/scala/com/gu/kindlegen/QuerierSpec.scala]: We tend to use `should be (value)` rather than `assert`. Check the tests within `Concierge` or `Porter` for examples. It allows for more expressive checking, see: http://www.scalatest.org/user_guide/using_matchers

File: src/main/scala/com/gu/kindlegen/Article.scala (hunk 8 above)
Comment (tsop14) [reply src/main/scala/com/gu/kindlegen/Article.scala]: As you are using a case class you do not need new to create an instance.

File: src/main/scala/com/gu/kindlegen/BookSectionPage.scala (hunk 11 above)
Comment (tsop14) [inline src/main/scala/com/gu/kindlegen/BookSectionPage.scala]: Same as Anne's comment above. What is a tsil?
